src/ext/Util/ca/scaexec.cpp

```diff
@@ -747,95 +820,136 @@ extern "C" UINT __stdcall CreateUser(
+                            }
+
+                            if (!(SCAU_ALLOW_LOGON_AS_BATCH & iOriginalAttributes) && (SCAU_ALLOW_LOGON_AS_BATCH & iAttributes))
+                            {
+                                iRollbackAttributes |= SCAU_ALLOW_LOGON_AS_BATCH;
+                            }
+                        }
+
+                        hr = WcaCaScriptWriteString(hRollbackScript, pUserInfo1->usri1_comment);
+                        ExitOnFailure(hr, "Failed to add rollback comment to rollback script.");
+
+                        if (!pUserInfo1->usri1_comment || !*pUserInfo1->usri1_comment)
+                        {
+                            iRollbackAttributes |= SCAU_REMOVE_COMMENT;
+                        }
+
+                        hr = WcaCaScriptWriteNumber(hRollbackScript, iRollbackAttributes);
+                        ExitOnFailure(hr, "Failed to add rollback attributes to rollback script.");
+
+                        // Nudge the system to get all our rollback data written to disk.
+                        WcaCaScriptFlush(hRollbackScript);
+                    }
+                }
+
+                ExitOnNetworkError(::SetUserPassword(pwzServerName, pwzName, pwzPassword));
 
-                    er = ::NetUserSetInfo(wz, pwzName, 1, reinterpret_cast<LPBYTE>(puserInfo), &dw);
+                if (SCAU_REMOVE_COMMENT & iAttributes)
+                {
+                    ExitOnNetworkError(SetUserComment(pwzServerName, pwzName, L""));
                 }
+                else if (pwzComment && *pwzComment)
+                {
+                    ExitOnNetworkError(SetUserComment(pwzServerName, pwzName, pwzComment));
+                }
+
+                DWORD flags = pUserInfo1->usri1_flags;
+
+                hr = ApplyAttributes(iAttributes, &flags);
+                ExitOnFailure(hr, "failed to apply attributes");
+
+                ExitOnNetworkError(SetUserFlags(pwzServerName, pwzName, flags));
             }
-            else if (!(SCAU_FAIL_IF_EXISTS & iAttributes))
+            else
             {
-                er = NERR_Success;
+                // The user exists, but was not updated
+                if (SCAU_REMOVE_COMMENT & iAttributes)
+                {
+                    ExitOnNetworkError(SetUserComment(pwzServerName, pwzName, L""));
+                }
+                else if (pwzComment && *pwzComment)
+                {
+                    ExitOnNetworkError(SetUserComment(pwzServerName, pwzName, pwzComment));
+                }
+
+                if (SCAU_FAIL_IF_EXISTS & iAttributes)
+                {
+                    ExitOnNetworkError(NERR_UserExists);
+                }
             }
         }
         else if (NERR_PasswordTooShort == er || NERR_PasswordTooLong == er)
         {
             MessageExitOnFailure(hr = HRESULT_FROM_WIN32(er), msierrUSRFailedUserCreatePswd, "failed to create user: %ls due to invalid password.", pwzName);
         }
+
+    LNExit:
```

Comment:
If you really need a jump, then pull out the necessary code to a separate function and use `LExit` normally. However, I _think_ it makes sense to try updating as much as possible so this extra label might not be necessary. As noted above, definitely don't want multiple jump points in a function.

Reply:
No jump needed.